plenum/test/bls/helper.py

```diff
@@ -174,6 +232,30 @@ def change_bls_key(looper, txnPoolNodeSet,
     return new_blspk
 
 
+def sdk_change_bls_key(looper, txnPoolNodeSet,
+                       node,
+                       sdk_pool_handle,
+                       sdk_wallet_steward,
+                       add_wrong=False,
+                       new_bls=None):
+    new_blspk = init_bls_keys(node.keys_dir, node.name)
+    key_in_txn = new_bls or new_blspk \
+        if not add_wrong \
+        else base58.b58encode(randomString(128).encode())
+    node_dest = hexToFriendly(node.nodestack.verhex)
+    sdk_send_update_node(looper, sdk_wallet_steward,
+                         sdk_pool_handle,
+                         node_dest, node.name,
+                         None, None,
+                         None, None,
+                         bls_key=key_in_txn,
+                         services=None)
+    waitNodeDataEquality(looper, node, *txnPoolNodeSet[:-1])
```

Comment:
you pass some node to the func, it could be any node from txnPoolNodeSet, but to check equality you exclude always the last one.
I think you need to exclude the node you pass to func but not the last one

Reply:
Done